include/deal.II/base/polynomials_hermite.h

```diff
@@ -0,0 +1,195 @@
+ * r-th derivative, for polynomial degree 2r+1).
+ */
+
+/**
+ * @addtogroup Polynomials
+ * @{
+ */
+
+namespace Polynomials
+{
+  /**
+   * This class implements Hermite interpolation polynomials
+   * enforcing the maximum posible level of regularity in the
+   * FEM basis, given the polynomial degree. This can only
+   * result in odd degree bases, since there must be the same
+   * numer of functions representing shape value and derivatives
+   * at $y=0$ and $y=1$ on the reference interval $y \in [0,1]$.
+   *
+   * Indices $0 \leq j \leq q$ refer to polynomials corresponding
+   * to a non-zero derivative (or shape value for $j=0$) of
+   * order $j$ at $x=0$, and indices $q+1 \leq j \leq 2q+1$
+   * refer to polynomials with a non-zero derivative of order
+   * $j-(q+1)$ (or value for $j=q+1$) at $x=1$.
+   *
+   * In the current implementation, the basis is created by
+   * solving a triangular matrix system by using Bernstein
+   * polynomials. The basis is then rescaled so that a function
+   * corresponding to a non-zero $r^{th}$ derivative has
+   * derivative value $r! 4^{r}$ at the corresponding node.
+   */
+  class HermitePoly : public Polynomial<double>
+  {
+  public:
+    /**
+     * Constructor that takes an order @p regularity to enforce
+     * regularity up to and an index @p index for the Hermite
+     * polynomial to be returned. The resulting polynomial will
+     * have a degree of $2 \times$ @p regularity $+1$ and
+     * obey the following conditions:
+     *
+     * @f{align*}{
+     * &\begin{matrix}
+     *   \left. \frac{d^{i}}{dy^{i}} H(y) \right\vert_{y=0}
+     *          = i! 4^{i} \delta_{i, index}, \hfill
+     *          &\qquad \hfill 0 \leq i \leq regularity, \\
+     *   \left. \frac{d^{i}}{dy^{i}} H(y) \right\vert_{y=1}
+     *          = 0, \hfill &\qquad \hfill 0 \leq i \leq regularity,
+     * \end{matrix} \qquad 0 \leq index \leq regularity, \\
+     * &\begin{matrix}
+     *  \left. \frac{d^{i}}{dy^{i}} H(y) \right\vert_{y=0}
+     *          = 0, \hfill &\qquad \hfill 0 \leq i \leq regularity, \\
+     * \left. \frac{d^{i}}{dy^{i}} H(y) \right\vert_{y=1}
+     *          = i! 4^{i} \delta_{i, index-(regularity+1)}, \hfill
+     *          &\qquad \hfill 0 \leq i \leq regularity,
+     *\end{matrix} \qquad regularity+1 \leq index
+     *          \leq 2 \times regularity + 1,
+     * @f}
+     *
+     * where $\delta_{i,j}$ is equal to $1$ whenever $i=j$,
+     * and equal to $0$ otherwise.
+     *
+     * @param regularity
+     * @param index
+     */
+    HermitePoly(const unsigned int regularity, const unsigned int index)
+      : Polynomial<double>(hermite_poly_coeffs(regularity, index))
+      , degree(2 * regularity + 1)
+      , regularity(regularity)
+      , side_index(index % (regularity + 1))
+      , side((index > regularity) ? 1 : 0){};
+
+
+
+    /**
+     * This function is used internally to generate a vector of Polynomial

```

Comment:
This is used by fe_hermite rather than for use by an end user, but you're right that this is badly worded. I will change the description to better reflect its purpose in the code.